mappings/net/minecraft/item/BlockItem.mapping

```diff
@@ -7,10 +7,23 @@ CLASS bad net/minecraft/item/BlockItem
 		ARG 1 player
 		ARG 2 pos
 		ARG 3 stack
+	METHOD a getPlaceSound (Lbvj;)Lyj;
+	METHOD a addProperty (Lbvj;Lbwm;Ljava/lang/String;)Lbvj;
+		ARG 0 state
+		ARG 1 property
+		ARG 2 name
 	METHOD a afterBlockPlaced (Lev;Lbhh;Lavx;Lbca;Lbvj;)Z
+	METHOD a place (Lev;Lbhh;Lbca;Lbvj;)Lbvj;
+		ARG 1 pos
+		ARG 2 world
+		ARG 3 stack
+		ARG 4 state
 	METHOD a registerBlockItemMap (Ljava/util/Map;Lbbv;)V
 		ARG 1 map
 	METHOD b getPlacementContext (Lbae;)Lbae;
 	METHOD b canPlace (Lbae;Lbvj;)Z
+		ARG 1 context
+		ARG 2 state
 	METHOD c getBlockState (Lbae;)Lbvj;
+	METHOD d checkIfStateAllowsPlacement ()Z
```

Comment:
`stateAllowsPlacement`, maybe even `allowsPlacement` or `canBePlaced`

Reply:
No, this returns true if the placement code should check if the state allows placement. Maybe it could be named `shouldCheckIfStateAllowsPlacement`?

Reply:
yes that works